src/core/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,184 @@
+import React, { Component, forwardRef } from 'react';
+import { default as styled } from 'styled-components';
+import classnames from 'classnames';
+import { SuomifiThemeProp, SuomifiThemeConsumer } from '../theme';
+import { baseStyles } from './Pagination.baseStyles';
+import { HtmlSpan, HtmlNav, HtmlDiv } from '../../reset';
+import { asPropType } from '../../utils/typescript';
+import { PageInput } from './PageInput/PageInput';
+import { Button } from '../Button/Button';
+
+const externalClassName = 'fi-link--external';
+
+export interface PaginationInputProps {
+  /** Page input action button label for screen readers  */
+  pageInputButtonText: string;
+  /** Input placeholder text */
+  inputPlaceholderText: string;
+  /** Error text shown with invalid page number input */
+  invalidValueErrorText: string;
+  /** Visually hidden label for input field */
+  labelText: string;
+}
+
+interface InternalPaginationProps {
+  /** Custom classname to extend or customize */
+  className?: string;
+  /** Is there any need to have as prop */
+  asProp?: asPropType;
+  /** aria-label for pagination  */
+  'aria-label': string;
+  /** Function for page number indicator text */
+  textFunction?: (currentPage: number, lastPage: number) => string;
+  /** Use small screen styling */
+  smallScreen?: boolean;
+  /** Number of the current page */
+  currentPage: number;
+  /** Number of the last page */
+  lastPage: number;
+  /** Returns the selected page number */
+  onChange: (page: number) => void;
+
+  /** Ref is forwarded to root element. Alternative for React `ref` attribute. */
+  forwardedRef?: React.RefObject<HTMLElement>;
+  /** Next page button label for screen readers  */
+  ariaNextButtonLabel: string;
+  /** Previous page button label for screen readers */
+  ariaPreviousButtonLabel: string;
+
+  pageInput?: boolean;
+
+  paginationInputProps?: PaginationInputProps;
```

Comment:
if the name of the element is `pageInput`, would `pageInputProps` make sense here?